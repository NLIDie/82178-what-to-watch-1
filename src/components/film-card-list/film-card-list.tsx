import * as React from "react";

// Components
import {FilmCard} from "_Components/film-card/film-card";

// Types
import {TProps} from "./film-card-list.types";
import {TFilm} from "_Types/film";

export function FilmCardList(props: TProps) {
  const {films} = props;

  const filmsCards = films.map(
      (film: TFilm): JSX.Element => (
        <FilmCard
          key={film.id}
          id={film.id}
          title={film.name}
          poster={film.previewImage}
          previewVideo={film.previewVideoLink}
        />
      )
  );

  return <div className="catalog__movies-list">{filmsCards}</div>;
}
