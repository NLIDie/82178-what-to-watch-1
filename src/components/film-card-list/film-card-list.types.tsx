// Types
import {TFilm} from "_Types/film";

export type TProps = {
  films: TFilm[];
};
