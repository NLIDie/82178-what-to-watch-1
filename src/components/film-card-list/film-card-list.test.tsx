import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {FilmCardList} from "./film-card-list";

describe(`Components/FilmCardList`, () => {
  it(`Correct renderer the component`, () => {
    const tree = create(<FilmCardList films={[]} />);

    expect(tree).toMatchSnapshot();
  });
});
