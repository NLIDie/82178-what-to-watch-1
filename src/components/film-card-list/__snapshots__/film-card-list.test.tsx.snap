// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FilmCardList Correct renderer the component 1`] = `
<div
  className="catalog__movies-list"
/>
`;
